lib/model.js

```diff
@@ -2102,10 +2102,12 @@ class Model {
 
         // set createdAt/updatedAt attributes
         if (createdAtAttr && !values[createdAtAttr]) {
-          values[createdAtAttr] = now;
+          let createdAtField = instance.rawAttributes[createdAtAttr].field;
+          values[createdAtField || createdAtAttr] = now;
         }
         if (updatedAtAttr && !values[updatedAtAttr]) {
-          values[updatedAtAttr] = now;
+          let updatedAtField = instance.rawAttributes[updatedAtAttr].field;
+          values[updatedAtField || updatedAtAttr] = now;
         }
```

Comment:
@mickhansen this code block depend upon `values`, which are generated by `mapValueFieldNames` call. It can't be moved before that.